cmd/tusd/cli/log.go

```diff
@@ -16,22 +16,37 @@ func SetupStructuredLogger() {
 		level = slog.LevelDebug
 	}
 
-	slogger := slog.New(
-		slog.NewTextHandler(logWriter{stdout}, &slog.HandlerOptions{
+	replaceAttrFunc := func(groups []string, a slog.Attr) slog.Attr {
+		// Remove time attribute, because that is handled by the logger
+		if a.Key == slog.TimeKey {
+			return slog.Attr{}
+		}
+		// Rename `msg` to `event`
+		if a.Key == slog.MessageKey {
+			a.Key = "event"
+		}
+		return a
+	}
+
+	var handler slog.Handler
+	switch Flags.OutputFormat {
+	case "json":
+		handler = slog.NewJSONHandler(logWriter{log.New(os.Stdout, "", 0)}, &slog.HandlerOptions{
 			Level: level,
-			ReplaceAttr: func(groups []string, a slog.Attr) slog.Attr {
-				// Remove time attribute, because that is handled by the logger
-				if a.Key == slog.TimeKey {
-					return slog.Attr{}
-				}
-				// Rename `msg` to `event`
-				if a.Key == slog.MessageKey {
-					a.Key = "event"
-				}
-				return a
-			},
-		}))
-	slog.SetDefault(slogger)
+		})
+	default:
+		handler = slog.NewTextHandler(logWriter{log.New(os.Stdout, "", log.LstdFlags|log.Lmicroseconds)}, &slog.HandlerOptions{
+			Level:       level,
+			ReplaceAttr: replaceAttrFunc,
+		})
+	}
+
+	slog.SetDefault(slog.New(handler))
+
+	if Flags.OutputFormat == "json" {
+		stdout = log.Default()
```

Comment:
Could you explain these lines?

Reply:
Comment added, these are used in the cli package.

Reply:
If I would not do this, non of the output of the cli package would be in json. The goal was to ensure all output is json.